great_expectations/profiler/domain_builder/util.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any, Dict, List, cast
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.domain_types import SemanticDomainTypes
+from great_expectations.validator.validation_graph import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+def translate_table_column_type_to_semantic_domain_type(
+    validator: Validator, column_name: str
+) -> SemanticDomainTypes:
+    # Note: As of Python 3.8, specifying argument type in Lambda functions is not supported by Lambda syntax.
+    column_types_dict_list: List[Dict[str, Any]] = list(
+        filter(
+            lambda column_type_dict: column_name in column_type_dict,
+            validator.get_metric(
+                metric=MetricConfiguration(
+                    metric_name="table.column_types",
+                    metric_domain_kwargs={},
+                    metric_value_kwargs={
+                        "include_nested": True,
+                    },
+                    metric_dependencies=None,
+                )
+            ),
+        )
+    )
+    if len(column_types_dict_list) != 1:
+        raise ge_exceptions.ProfilerExecutionError(
+            message="A unique column could not be found while obtaining semantic type information."
```

Comment:
I wonder if the word `unique` here might be a bit confusing. Maybe a message more like f"While obtaining semantic type information, we found {len(column_types_dict_list)} column/s. Please ensure that the specified column name refers to exactly one column."

Reply:
@talagluck I like your suggestion!  In the next commit.